### PR TITLE
New UI for receive payment flows

### PR DESCRIPTION
**Pull request by erdemyerebasmaz**
This PR unifies receive payment flows under the same page.

### Changelist:
- Unify receive payment methods under same UI d8194ec
  - All receive payment pages are moved under `receive_payment` dir and `ReceivePaymentPage`, which takes a `pageIndex` and displays the selected payment methods page. Renamed files to reflect these changes.
  - UI's are made consistent across all payment pages where possible. 
    - All pages now use the same `AddressWidget` to display payment information.
    - They use similar fee information widgets etc.
  - Dialogs on payment flows are removed or integrated to a fulll page UI as part of their respective flow.
- Open the bottom sheet menu item via ReceivePaymentPage 3e586b7
- Remove payment method selector 69a9505
#### Layout & Theme issues:
- Do not override ExpiryAndFeeMessage colors on light theme  edcc375
- Move padding to ReceivePayment page 241068a
- Fix share & copy icon colors 077ded2
- Fix loader alignment on receive pages 8dc0e40
- Fix header displacement after QR code loads fc40ccb

#### Misc:
- Added Breez icon to QR image
- Replace hardcoded messages from Breez-Translations d7e6b25
- Add helper methods for readability d46d4da
  - These will be extracted into their own widgets
  
  
 ### TODO:
 - [x] Track payments and close the payment information to display bubble animation on home page when the payment is received 35db836

### CODE REVIEW

File: lib/routes/receive_payment/lightning/widgets/compact_qr_image.dart [hunk 1]
```diff
@@ -49,7 +49,7 @@ class CompactQRImage extends StatelessWidget {
   final double? size;
   final bool bip21;
 
-  const CompactQRImage({super.key, required this.data, this.size, this.bip21 = false});
+  const CompactQRImage({super.key, required this.data, this.size, this.bip21 = true});
```
Comment (dangeross) [inline lib/routes/receive_payment/lightning/widgets/compact_qr_image.dart:52]: Assuming this is a bolt11 invoice (lightning widget), should bip21 be set to `false`?

File: lib/routes/receive_payment/onchain/bitcoin_address/receive_bitcoin_address_payment_page.dart [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+  FutureBuilder<ReceivePaymentResponse> _buildQRCode() {
+    final texts = context.texts();
+    return FutureBuilder(
+      future: receivePaymentResponse,
+      builder: (BuildContext context, AsyncSnapshot<ReceivePaymentResponse> snapshot) {
+        return AddressWidget(
+          snapshot: snapshot,
+          title: texts.withdraw_funds_btc_address,
+          type: AddressWidgetType.lightning,
```
Comment (dangeross) [inline lib/routes/receive_payment/onchain/bitcoin_address/receive_bitcoin_address_payment_page.dart]: Just checking, should this be `AddressWidgetType.bitcoin`?

File: lib/routes/receive_payment/onchain/liquid_address/receive_liquid_address_payment_page.dart [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+import 'package:l_breez/widgets/flushbar.dart';
+import 'package:l_breez/widgets/keyboard_done_action.dart';
+import 'package:l_breez/widgets/loader.dart';
+import 'package:l_breez/widgets/single_button_bottom_bar.dart';
+import 'package:l_breez/widgets/transparent_page_route.dart';
+import 'package:service_injector/service_injector.dart';
+import 'package:share_plus/share_plus.dart';
+
+class ReceiveLiquidAddressPaymentPage extends StatefulWidget {
+  static const routeName = "/receive_bitcoin_address";
```
Comment (dangeross) [inline lib/routes/receive_payment/onchain/liquid_address/receive_liquid_address_payment_page.dart]:
```suggestion
  static const routeName = "/receive_liquid_address";
```

File: lib/routes/receive_payment/onchain/liquid_address/receive_liquid_address_payment_page.dart [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+              child: Padding(
+                padding: const EdgeInsets.fromLTRB(32, 0, 32, 0),
+                child: Text(
+                  texts.reverse_swap_upstream_generic_error_message(snapshot.errorMessage),
+                  textAlign: TextAlign.center,
+                ),
+              ),
+            );
+          }
+          if (snapshot.onchainPaymentLimits == null) {
```
Comment (dangeross) [inline lib/routes/receive_payment/onchain/liquid_address/receive_liquid_address_payment_page.dart]: I know we're not using it, but we don't need onchainPaymentLimits for Liquid

File: lib/routes/receive_payment/widgets/address_widget/address_qr_widget.dart [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        error: snapshot?.error,
+        displayErrorMessage: snapshot?.error != null
+            ? extractExceptionMessage(snapshot!.error!, texts)
+            : texts.qr_code_dialog_warning_message_error,
+      ),
+      secondChild: address == null && snapshot?.data == null
+          ? const SizedBox.shrink()
+          : Column(
+              children: [
+                AddressQR(bolt11: address ?? snapshot!.data!.destination, bip21: true),
```
Comment (dangeross) [inline lib/routes/receive_payment/widgets/address_widget/address_qr_widget.dart:42]: Should the `bip21` value depend on the AddressWidgetType?

File: lib/routes/receive_payment/widgets/address_widget/address_qr.dart [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import 'package:flutter/material.dart';
+import 'package:l_breez/routes/receive_payment/lightning/widgets/compact_qr_image.dart';
+
+class AddressQR extends StatelessWidget {
+  final String bolt11;
```
Comment (dangeross) [inline lib/routes/receive_payment/widgets/address_widget/address_qr.dart:5]: Maybe rename this to `destination` or `data`

File: lib/routes/receive_payment/onchain/liquid_address/receive_liquid_address_payment_page.dart (hunk 3 above)
Comment (erdemyerebasmaz) [reply lib/routes/receive_payment/onchain/liquid_address/receive_liquid_address_payment_page.dart]: Removed `ReceiveLiquidAddressPaymentPage` as it's unused 56df30ea90ce63a171c86d312807e7b23d96db4f

File: lib/routes/receive_payment/onchain/bitcoin_address/receive_bitcoin_address_payment_page.dart (hunk 2 above)
Comment (erdemyerebasmaz) [reply lib/routes/receive_payment/onchain/bitcoin_address/receive_bitcoin_address_payment_page.dart]: Addressed as part of: [Unify fee & limits message box widgets](https://github.com/breez/misty-breez/pull/167/commits/75bccf2f0c3639fec6a44826c0dd80367db61edd)

File: lib/routes/receive_payment/widgets/address_widget/address_qr.dart (hunk 6 above)
Comment (erdemyerebasmaz) [reply lib/routes/receive_payment/widgets/address_widget/address_qr.dart:5]: I agree, these widgets will be renamed & refactored next.

Comment (erdemyerebasmaz) [reply lib/routes/receive_payment/widgets/address_widget/address_qr.dart]: Done. [Refactor AddressWidget](https://github.com/breez/misty-breez/pull/167/commits/ea1ef395a50d2e950f0615a8653dbfb07a807abd)

File: lib/routes/receive_payment/widgets/address_widget/address_qr_widget.dart (hunk 5 above)
Comment (erdemyerebasmaz) [reply lib/routes/receive_payment/widgets/address_widget/address_qr_widget.dart]: Removed uppercase QR trick to produce smaller QR codes.

**Comment by erdemyerebasmaz** [comment]
@dangeross New changes since last review:

#### Payment tracking & successful payment animation
- Track payments and pop to home page to display bubble animation on receiving payment 
- Only track incoming payments
- Fade out the bubble animation and pop the SuccessfulPaymentRoute 
- Display payment received message upon receiving a payment
- Added comments for flip transition animation
- Removed `didUpdateWidget` to avoid displaying `FlipTransition` twice

#### DestinationWidget changes
- Refactored `AddressWidget`
- Add padding to payment limits message box loader
- Renamed and moved address widget folders

#### LNURL payment
- Display errors for LNURL payments that fall below payment limits
- Fix the minimum value pasted on clicking limits label on LNURL payment page
- Use domain from `LnUrlPayRequestData` on it's title

#### LNURL Withdraw
- Support LNURL Withdraw with new UI 
- Created a dedicated LnUrl Withdraw page
- Display errors for URLs recognized through QR scanner
- Fix amount auto focus on `LnUrlWithdrawPage`
- Remove LnUrlWithdraw logic from `ReceiveLightningPaymentPage`

File: lib/routes/receive_payment/lnurl/lnurl_withdraw_page.dart [hunk 7]
```diff
@@ -0,0 +1,344 @@
...
+    final data = widget.requestData;
+
+    return BlocBuilder<CurrencyCubit, CurrencyState>(
+      builder: (context, currencyState) {
+        final isFixedAmount = data.minWithdrawable == data.maxWithdrawable;
+        final minWithdrawableFormatted = currencyState.bitcoinCurrency.format(minWithdrawable);
+        final maxWithdrawableFormatted = currencyState.bitcoinCurrency.format(maxWithdrawable);
+
+        final isFixedAmountWithinLimits = minWithdrawable == maxWithdrawable;
+        if (!isFixedAmountWithinLimits) {
```
Comment (dangeross) [inline lib/routes/receive_payment/lnurl/lnurl_withdraw_page.dart:193]: Is this right? Seems inconsistent with the error text and the min/max is already checked

Comment (erdemyerebasmaz) [reply lib/routes/receive_payment/lnurl/lnurl_withdraw_page.dart]: Explained in Slack and [added a different message if fixed payment amount is outside allowed limits](https://github.com/breez/misty-breez/pull/167/commits/ab87441912a0365c837c9e250447fd3db6ad5b35)

File: lib/routes/receive_payment/lnurl/lnurl_withdraw_page.dart [hunk 8]
```diff
@@ -44,36 +42,75 @@ class LnUrlWithdrawPageState extends State<LnUrlWithdrawPage> {
...
+    final minWithdrawableSat = widget.requestData.minWithdrawable.toInt() ~/ 1000;
+    final maxWithdrawableSat = widget.requestData.maxWithdrawable.toInt() ~/ 1000;
+    final effectiveMinSat = max(response.receive.minSat.toInt(), minWithdrawableSat);
+    final effectiveMaxSat = min(response.receive.maxSat.toInt(), maxWithdrawableSat);
+    if (effectiveMaxSat < effectiveMinSat) {
+      final texts = context.texts();
+      final currencyCubit = context.read<CurrencyCubit>();
+      final currencyState = currencyCubit.state;
+
+      final isFixedAmountWithinLimits = _isFixedAmount && (effectiveMinSat == effectiveMaxSat);
```
Comment (dangeross) [inline lib/routes/receive_payment/lnurl/lnurl_withdraw_page.dart:84]: I think `_isFixedAmount` has not been set yet as `_initializePageData()` is called after

**Review by roeierez — APPROVED** [review]

**Pull request merged**